crates/pathfinder/src/state/merkle_tree.rs

```diff
@@ -463,6 +527,50 @@ impl<T: NodeStorage> MerkleTree<T> {
         Ok(result)
     }
 
+    /// TODO: Comment
+    pub fn get_proof(&self, key: &BitSlice<Msb0, u8>) -> anyhow::Result<Vec<ProofNode>> {
+        if self.root.borrow().is_empty() {
+            // TODO: Should be an error here since it's impossible to prove `key` of a non-existent tree.
+            // Plus, an empty vector means there is only a single edge.
+            return Ok(Vec::new());
+        }
+
+        let mut current = self.root.clone();
+        let mut height = 0;
+        let mut nodes: Vec<ProofNode> = Vec::new();
+        loop {
+            use Node::*;
+
+            let current_tmp = current.borrow().clone();
+
+            let next = match current_tmp {
+                Unresolved(hash) => {
+                    let node = self.resolve(hash, height)?;
+                    current.swap(&RefCell::new(node));
+                    current
+                }
+                Binary(binary) => {
+                    height += 1;
+                    let direction = binary.direction(key);
+                    let proof_node = ProofNode::Binary(BinaryProofNode::from(&binary));
+                    nodes.push(proof_node);
+
+                    binary.get_child(direction)
+                }
+                Edge(edge) if edge.path_matches(key) => {
+                    height += edge.path.len();
+                    let proof_node = ProofNode::Edge(EdgeProofNode::from(&edge));
+                    nodes.push(proof_node);
+                    edge.child
+                }
+                Leaf(_) | Edge(_) => {
+                    return Ok(nodes);
+                }
+            };
+            current = next;
+        }
```

Comment:
How much of this is simply the same as `traverse()`? Could we not simply make a call to `traverse` and then map the resulting nodes to the `PoofNode` types?